### PR TITLE
CLN: Static types in `pandas/_lib/lib.pyx`

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -103,12 +103,10 @@ def memory_usage_of_objects(arr: object[:]) -> int64_t:
 
     Does not include the actual bytes of the pointers
     """
-    i: Py_ssize_t
-    n: Py_ssize_t
-    size: int64_t
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:108]: we put this in because it is consistent with pure-python syntax

File: pandas/_libs/lib.pyx [hunk 2]
```diff
@@ -1000,36 +998,37 @@ cdef inline bint c_is_list_like(object obj, bint allow_sets):
...
-    'bool': 'boolean',
-    'b': 'boolean',
-    'datetime64[ns]': 'datetime64',
-    'M': 'datetime64',
-    'timedelta64[ns]': 'timedelta64',
-    'm': 'timedelta64',
-    'interval': 'interval',
-}
+cdef:
+    dict _TYPE_MAP = {
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:1002]: do we actually get anything from typing this?

one of the pie-in-the-sky goals is to get the cython code close enough to valid python that we can make flake8 work on it; this is going in the other direction

Comment (MomIsBestFriend) [reply pandas/_libs/lib.pyx:1002]: > do we actually get anything from typing this?

I looked at the generated C code, and it looks like we don't gain anything from typing this, I assumed we did, but I guess not.

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

**Pull request merged**